engine/src/QuadTree.cpp

```diff
@@ -0,0 +1,239 @@
+//
+// Created by tdancois on 7/31/16.
+//
+
+#include <SFML/Graphics/Text.hpp>
+#include <sstream>
+#include "QuadTree.h"
+
+QuadTree::QuadTree(const unsigned int &x,
+                   const unsigned int &y,
+                   const unsigned int &width,
+                   const unsigned int &height,
+                   const unsigned int &node_capacity_,
+                   const unsigned int &level_) :
+        node_capacity(node_capacity_),
+        level(level_),
+        isSplit(false) {
+
+    shape.setPosition(x, y);
+
+    sf::Vector2f size(width, height);
+    shape.setSize(size);
+
+    shape.setFillColor(sf::Color::Transparent);
+    shape.setOutlineColor(sf::Color::Cyan);
+    shape.setOutlineThickness(2);
+}
+
+QuadTree::~QuadTree() {
+    northEast.reset(nullptr);
```

Comment:
You don't need to do that. The object will be destroyed automatically. The main advantage of using smart pointers is that you don't manage the memory manually